libraries/AP_Airspeed/AP_Airspeed.h

```diff
@@ -53,39 +64,83 @@ class AP_Airspeed
     float get_airspeed(uint8_t i) const {
         return state[i].airspeed;
     }
-    float get_airspeed(void) const { return get_airspeed(primary); }
+    float get_airspeed(void) const {
+        if (blend_state.active) {
+            return blend_state.airspeed;
+        } else {
+            return get_airspeed(primary);
+        }
+    }
 
     // return the unfiltered airspeed in m/s
     float get_raw_airspeed(uint8_t i) const {
         return state[i].raw_airspeed;
     }
-    float get_raw_airspeed(void) const { return get_raw_airspeed(primary); }
+    float get_raw_airspeed(void) const {
+        if (blend_state.active) {
+            return blend_state.raw_airspeed;
+        } else {
+            return get_raw_airspeed(primary);
+        }
+    }
 
     // return the current airspeed ratio (dimensionless)
     float get_airspeed_ratio(uint8_t i) const {
         return param[i].ratio;
     }
-    float get_airspeed_ratio(void) const { return get_airspeed_ratio(primary); }
+    float get_airspeed_ratio(void) const {
+        if (blend_state.active) {
+            return blend_state.ratio;
+        } else {
+            return get_airspeed_ratio(primary);
+        }
+    }
 
     // get temperature if available
     bool get_temperature(uint8_t i, float &temperature);
-    bool get_temperature(float &temperature) { return get_temperature(primary, temperature); }
+    bool get_temperature(float &temperature) {
+        if (blend_state.active) {
+            temperature = blend_state.temperature;
+            return true;
+        } else {
+            return get_temperature(primary, temperature);
+        }
+    }
 
     // set the airspeed ratio (dimensionless)
     void set_airspeed_ratio(uint8_t i, float ratio) {
         param[i].ratio.set(ratio);
     }
-    void set_airspeed_ratio(float ratio) { set_airspeed_ratio(primary, ratio); }
+    void set_airspeed_ratio(float ratio) {
```

Comment:
Rather then carrying a TODO here which sets the wrong field, the whole function should just be removed since it clearly doesn't get called anywhere. At a minimum it shouldn't be blindly overwriting a sub sensor if the caller expected it to be for the primrary instance.

Reply:
> since it clearly doesn't get called anywhere
If it does not - I will remove.